### ISSUE TITLE
Account Delegation not working correctly

### ISSUE THREAD

**Issue by oe4dns** [friendica/friendica#13944]
### Bug Description

Since the upgrade to Friendica 2023-12 I am experiencing issues with account delegation - the "Accounts" Menu is only available to switch into the "controlled" account, bot not to switch back to the admin. See screenshots for a more clear explanation.

![Select_Account](https://github.com/friendica/friendica/assets/1168512/95788678-cdec-45b7-8196-8352f8115272)

In the screenshot above, everything works as expected. There is an "Accounts" menu entry that I can use to change the account that I want to use.

![Configure_Delegates](https://github.com/friendica/friendica/assets/1168512/09686f7f-865e-419a-9963-2717953ba422)

This shows the sub-account that has access delegated to another. The Problem is, there is no "Accounts" Menu entry to switch back to the admin account. This worked before 2023-12 but stopped working after the upgrade. The only way to get back to the original account is to log out and back in.

### Platform Info

Friendica Version:

2023-12

Friendica Source: 

PHP version:

8.2.7

SQL version:

mariadb 10.11.6

**Comment by oe4dns** [comment]
Just upgraded to Friendica 2024-03 - The issue is still there.

**Comment by MrPetovan** [comment]
Thank you for the update!

**Comment by informapirata** [comment]
> Thank you for the update!

Sorry, but I think @oe4dns oe4dns is doing something wrong.

The link referred to in the first screenshot is https://fed.sonnenmulde.at/delegation

The link referred to in the second screenshot is
https://fed.sonnenmulde.at/settings/delegation

**Comment by oe4dns** [comment]
I am certainly not!

The second image shows that there indeed is an acitve delebation. But more importantly, the dropdown menu is missing the "Accounts" entry between "Contacts" and "Directory"

So there is no possibility to reach https://fed.sonnenmulde.at/delegation from there. And if I input the path manually, it is just not available.

**Comment by oe4dns** [comment]
FYI, 2024-08 Problem still persists.

**Comment by oe4dns** [comment]
2024-12 same, same

**Comment by marcusxss** [comment]
Side note: How about just always showing "Accounts" in the menu, no matter whether the user has multiple accounts (identities) yet or not? (edited)

**Comment by sekretaerbaer** [comment]
Why should you display something that the user can't use, it's confusing. 
And it doesn't solve the problem, if it is a problem at all. 

If I am the authorised representative for account B with account A, I can switch from account A (if I am logged in with account A) to account B and back again. 

However, if I am logged in with account B, I cannot logically switch to account A because account B is not the authorised representative of account A.

This only applies if account A and B are two independent accounts and not ‘additional accounts’.

The difference is that ‘additional accounts’ have the same e-mail address and two independent accounts also have two different e-mail addresses.

I have tested this on my instances.

**Comment by marcusxss** [comment]
~~Okay, the main issue here, as I understand it, is:~~
~~You can switch to a delegate account but not back~~

~~I would expect that one's login cookie remains the same even after switching accounts (as opposed to signing directly into a delegate), so Friendica knows which account you originally signed in as, and thus which accounts you have access to?~~

Edit: Never mind the above - from what you describe that part indeed seems to work as expected...?

My PR, on second thought, would not address this issue, but only ensure that "Accounts" is always visible in the user menu. (The second screenshot shows "Accounts" is missing there.)
The user can always use that page to: 
1. Create other accounts (incl. groups, pages)
2. Assign parent user(s)
...no matter whether they're already using the delegates feature or not, no?

**Comment by sekretaerbaer** [comment]
He would not be able to create other accounts (including groups and pages) via the /delegation page, nor would he be able to assign any parent users.
To do this, he would need to click on the link at the bottom of the ‘Manage your accounts’ page, which would redirect him to /settings/delegation, where he could then ‘register an additional account’.

**Comment by marcusxss** [comment]
You're right, it's one step further. It's a simpler route to get there than trawling through settings to find a core feature, though.

But maybe there's a better idea:  
It could link to /delegation if one already has delegates, or to /settings/delegation if not.

**Comment by sekretaerbaer** [comment]
However, this would be confusing for users because they would think that they could use the link to create additional accounts or delegate them, which they would not be able to do and would have to click on the link to /settings/delegation again. This would mean that users who already have additional accounts or delegations would suddenly have to click twice instead of once.

**Comment by marcusxss** [comment]
With the last suggestion, if they already have additional accounts or delegations it would take them to the page the link is going to right now. No additional clicks.

It *may* be confusing, though. But it's also confusing that something new pops up in the menu after a delegation is created, with no announcement of that being the case. 
Another reason is that I saw users on my instance create a guide for new users expecting that link to be there for them, but it wasn't. The ideal in this case would be that it's always there and always links to the same page. I think it's a common case that existing users will guide new users (friends/family) on how to use it, and months down the line they won't remember that that link only popped up after creating a delegation.

**Comment by oe4dns** [comment]
By chance I deacitvated 2 Factor Authentication on both my accounts (Parent User and Delegate) - The Accounts menu entriy is visible again and it works as expected.

So the issue is connected to 2FA although I have no clue which account "causes" it. My guess would be on the Delegate but this is just a guess.

**Comment by annando** [comment]
AFAIK there is some other open issue that was about 2FA as well. I guess that it is connected. I have to set up 2FA for that on my system to try it out, I guess.

**Comment by oe4dns** [comment]
At least the 2FA side of things (enabling/disabling ...) works as expected. I don't think many have tried disabling yet because thats not the normal usecase.
